### PR TITLE
Add aggregation client

### PR DESCRIPTION
**Pull request by corest**

### CODE REVIEW

**Comment by kopiczko** [comment]
I'd like to learn why is it needed

**Comment by corest** [comment]
It is needed for metrics-server service (https://github.com/giantswarm/kubernetes-metrics-server). There is `apiservice` api extension resource. So for proper testing of the chart deployment/migration - we need a client for aggregation api. This client is not yet a part of the main k8s client

**Review by kopiczko — APPROVED** [review]

File: pkg/framework/host.go [hunk 1]
```diff
@@ -86,6 +88,10 @@ func NewHost(c HostConfig) (*Host, error) {
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
+	k8sAggregationClient, err := aggregationclient.NewForConfig(restConfig)
```
[the remark this answers is not in the document]
Comment (corest) [reply pkg/framework/host.go]: Why would it break smth?

[the remark this answers is not in the document]
Comment (MarcelMue) [reply pkg/framework/host.go]: Do we need some specific values in the `restConfig` here, or can you vendor this into e.g. `aws-operator` and expect it to be fine.

[the remark this answers is not in the document]
Comment (corest) [reply pkg/framework/host.go]: That's the same `restConfig`, which is used in k8s-client. I'm just adding one more client, not changing existing.

But when there is something about dependencies - I never expect it to be fine. But I need this client for `metrics-server` migration code. Is there any other way to work with `apiservices` ?

**Review by xh3b4sd — APPROVED** [review]

File: pkg/framework/host.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcelMue) [reply pkg/framework/host.go]: If it's the same as in `k8sClient`, then this should be fine.

**Review by MarcelMue — APPROVED** [review]

**Review by rossf7 — APPROVED** [review]
Changes LGTM and this isn't a breaking change so no blocker. 

 I share the deps concern because we use e2e-harness in a lot of places but dep prune should help here.

We also have `e2eclients` library but at the moment it only has cloud provider clients. https://github.com/giantswarm/e2eclients

Just wanted to point out the alternative.

**Comment by corest** [comment]
aggregation API is going to be part of the general client, so it will be removed with next or after next release. But for now, that's a blocker for metrics-server

**Pull request merged**